### PR TITLE
[redis_proxy] Add support for SELECT and KEYS

### PR DESCRIPTION
**Pull request by duanhongyi**
This RP mainly improves the compatibility of redis proxy, improvement as follows:

* Add support for the select command, referring to the implementation of Apache KVRocks and Microsoft Garnet, simply returns OK.
* Add keys command support, send keys commands to each shard on the backend, and then merge the results.
* For unsupport commands, the return value is the same as Redis, because some clients judge this return value, for example, [Java lettuce client](https://github.com/redis/lettuce/blob/14de5b88f922581f996c3b3311cb253eb36b32db/src/main/java/io/lettuce/core/RedisHandshake.java#L328)

### CODE REVIEW

**Review by adisuissa — COMMENTED** [review]
Assigning @weisisea @mattklein123 as codeowners
/assign @weisisea @mattklein123

File: source/extensions/filters/network/redis_proxy/conn_pool_impl.cc [hunk 1]
```diff
@@ -69,6 +69,10 @@ void InstanceImpl::init() {
   });
 }
 
+//
+//
```
Comment (mattklein123) [inline source/extensions/filters/network/redis_proxy/conn_pool_impl.cc:73]: nit: ?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM at a high level. Can you fix the typo comments and then push to see if we can get CI green?

/wait

File: source/extensions/filters/network/redis_proxy/conn_pool_impl.cc (hunk 1 above)
Comment (duanhongyi) [reply source/extensions/filters/network/redis_proxy/conn_pool_impl.cc:73]: ok

**Comment by duanhongyi** [comment]
I rebased the last commit, mainly to solve the compatibility issues of the Java client.

https://github.com/envoyproxy/envoy/pull/37706/commits/234aa2390af88e8684956229d285c8f415e93269#diff-c0fa1fc35dc58d68142878191b27317c2de9789d87c14034fdb7ecfa7136226b

**Comment by duanhongyi** [comment]
There is still one that hasn't turned green, I will add some test cases later.

**Comment by duanhongyi** [comment]
Okay, all the checks have been passed.

@mattklein123

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**